internal/dag/dag.go

```diff
@@ -191,6 +191,14 @@ type UpstreamValidation struct {
 	SubjectName string
 }
 
+// CACertificateKey stores the key name for the TLS validation secret cert
+const CACertificateKey = "ca.crt"
+
+// ValidationContextCACert returns the CA certificate from ValidationContext
+func (vc *ValidationContext) ValidationContextCACert() []byte {
+	return vc.CACertificate.Object.Data[CACertificateKey]
+}
+
```

Comment:
Can you add the equivalent of `upstreamValidationSubjectAltName` as `GetSubjectName()`?

Reply:
Fixed.